syn/python/get_kge.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python3
+
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Parse a yosys area report and give a kGE equivalient
+
+import argparse
+
+
+def read_lib(lib_file_path):
+    with open(lib_file_path, 'r') as f:
+        lib_file = f.readlines()
+
+    cell_dict = {}
+    weighted_dict = {}
+    cell_name = None
+    for line_idx, line in enumerate(lib_file):
+        if line.startswith('  cell ('):
+            if cell_name is not None:
+                raise RuntimeError('{}:{} Found cell while searching for area'
+                                   .format(lib_file_path, line_idx + 1))
+            cell_name = line.split()[1].strip('()')
+        elif line.startswith('\tarea'):
+            if cell_name is None:
+                raise RuntimeError('{}:{} Found area while searching for cell'
+                                   .format(lib_file_path, line_idx + 1))
+            try:
+                cell_area = line.split()[2].strip(';')
+                cell_dict[cell_name] = float(cell_area)
+                cell_name = None
+            except (IndexError, ValueError):
+                raise RuntimeError('{}:{} Area declaration misformatted'
+                                   .format(lib_file_path, line_idx + 1))
+    for cell in cell_dict:
+        weighted_dict[cell] = cell_dict[cell] / cell_dict['NAND2_X1']
```

Comment:
Have the 1 GE cell specified as a config parameter rather than hardcoded? The name will probably differ between libraries.